filter.go

```diff
@@ -52,9 +52,15 @@ func Any(preds ...Predicate) Predicate {
 	}
 }
 
-// None returns true iff none of the preds are true
+// None returns true iff none of the preds are true,
+// or if no preds are passed in.
 func None(preds ...Predicate) Predicate {
-	p := Any(preds...)
+	var p Predicate = nil
+	if len(preds) == 0 {
+		p = All()
+	} else {
+		p = Any(preds...)
+	}
```

Comment:
I might rewrite this like so. And when I do, it breaks my brain to realize that `None() == Any()` :smile: 

```suggestion
	p := Any(preds...)
	if len(preds) == 0 {
		return p
	}
```

Reply:
If I squint, I can convince myself that `None()` should return all the resources in the manifest, because it's returning the complement all the predicates passed to it, and none of them match.

Reply:
Yup :) After our chat offline, I think Not() And() Or() are good, and making them accept only >= arg would be a nice QoL improvement as well.